### PR TITLE
Feat/deactivate privileges when state not live

### LINKED ISSUE #906: Home state change - Deactivate privileges if state is not live even if license data exists for state
We have identified an edge case where a user may move to a state that is not 'live' in the system, but has uploaded license data into the system. We don't practitioners to be able to move privileges over to the state if it is not live. In this case, the privileges should be deactivated as if there were no license for them in that state.

### Tasks
- [ ] TODO
- [ ] Automated tests
- [ ] API Docs
- [ ] Postman collection
- [ ] PR opened with labels / reviewers / assignee / linked-issue

### PR DESCRIPTION
**Pull request by landonshumway-ia**
We have identified an edge case where a user may move to a state that is not 'live' in the system, but has uploaded license data into the system. We don't practitioners to be able to move privileges over to the state if it is not live. In this case, the privileges should be deactivated as if there were no license for them in that state.

Closes #906 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of provider privileges and licenses during home jurisdiction changes with added checks for live jurisdictions and centralized privilege deactivation.

* **Tests**
  * Added a comprehensive test suite covering various provider home jurisdiction update scenarios, replacing the previous tests.

* **Chores**
  * Updated permissions to allow provider user management to read compact configuration data.
  * Enhanced smoke tests to support live jurisdiction configurations for more accurate home jurisdiction change testing.

### CODE REVIEW

File: backend/compact-connect/lambdas/python/common/cc_common/data_model/data_client.py [hunk 1]
```diff
@@ -1806,83 +1806,116 @@ def update_provider_home_state_jurisdiction(
...
-
-                # Get licenses from the current home state
-                current_home_state_licenses = provider_user_records.get_license_records(
-                    filter_condition=lambda license_data: license_data.jurisdiction == current_home_jurisdiction
+                # Check if the selected jurisdiction is live in the compact configuration
+                compact_config = self.config.compact_configuration_client.get_compact_configuration(compact)
+                is_jurisdiction_live = any(
+                    state['postalAbbreviation'].lower() == selected_jurisdiction.lower() and state.get('isLive', False)
+                    for state in compact_config.configuredStates
                 )
```
[the remark this answers is not in the document]
Comment (landonshumway-ia) [reply backend/compact-connect/lambdas/python/common/cc_common/data_model/data_client.py:1814]: The compact comes from a custom attribute on the user's cognito account directly, if there is a typo or failure pulling it up, we want to raise an internal exception so our alerting will fire

**Review by jlkravitz — APPROVED** [review]
looks good to me! @isabeleliassen good to merge.
